src/app/shared/services/location-map.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import leaflet from 'leaflet';
+
+@Injectable({
+  providedIn: 'root'
+})
```

Comment:
This should not be provided in `root`. You need a new instance of the service every time the component wants to use it. If you have 2 components on the same screen, they must not share this map object in the service. 
Instead, if you need to use the service, inject it in the component creation (`@Component(...)`). This way, every time the component is instantiated, it will get a new instance of the service.

```suggestion
@Injectable()
```